hooks/pre_gen_project.py

```diff
@@ -0,0 +1,14 @@
+import re
+import sys
+
+PROJECT_REGEX = r'^spyder_[_a-zA-Z0-9]+$'
+
+project_name = '{{ cookiecutter.project_name }}'
+
+if not re.match(PROJECT_REGEX, project_name):
+    print('ERROR: {}'.format(project_name),
+          'Project name should start with "spyder"',
+          'and be a valid Python module!')
+
```

Comment:
We should also validate the name of the repo or plugin

it should be 

Spyder-Something

or spyder-something